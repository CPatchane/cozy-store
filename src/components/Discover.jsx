import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'

import SmallAppItem from './SmallAppItem'

const DEMO_APPS = [
  {
    slug: 'bank',
    name: 'Bank',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-bank.svg')
  },
  {
    slug: 'sante',
    name: 'Sant√©',
    developer: {name: 'MAIF Ltd'},
    editor: '',
    icon: require('../assets/temp_apps/icon-sante.svg')
  },
  {
    slug: 'mon_logis',
    name: 'Mon Logis',
    developer: {name: 'HoodBrains'},
    editor: '',
    icon: require('../assets/temp_apps/icon-mon_logis.svg')
  },
  {
    slug: 'calendar',
    name: 'Calendar',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-calendar.svg')
  }
]

class Discover extends Component {
  render () {
    const { t } = this.props
    return (
      <div className='sto-discover sto-soon'>
        <h2 className='sto-discover-title'>{t('discover.title')}</h2>
        <div className='sto-discover-get-started'>
          <h3 className='sto-discover-get-started-title'>
            {t('discover.get_started')}
          </h3>
          <div className='sto-discover-get-started-list'>
            {DEMO_APPS.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
              />
            })}
          </div>
        </div>
      </div>
    )
  }
}

export default translate()(Discover)
