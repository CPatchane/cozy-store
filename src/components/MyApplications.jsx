import React, { Component } from 'react'

import { Route } from 'react-router-dom'
import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from './SmallAppItem'
import ApplicationModal from './ApplicationModal'

class MyApplications extends Component {
  constructor (props) {
    super(props)
    props.fetchApps()
  }

  onAppClick (appSlug) {
    this.props.history.push(`/myapps/${appSlug}/manage`)
  }

  render () {
    const { t, myApps, isFetching, error } = this.props
    return (
      <div className='sto-myapps'>
        <h2 className='sto-myapps-title'>{t('myapps.title')}</h2>
        <div className='sto-myapps-list'>
          {!isFetching && myApps && !!myApps.length &&
            myApps.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
                onClick={() => this.onAppClick(a.slug)}
              />
            })
          }
          {error &&
            <p>{error}</p>
          }
        </div>
        {isFetching &&
          <Spinner
            size='xxlarge'
            loadingType='appsFetching'
            middle='true'
          />
        }

        <Route path='/myapps/:appSlug/manage' render={({ match }) =>
          (<ApplicationModal {...this.props} match={match} />)
        } />
      </div>
    )
  }
}

export default translate()(MyApplications)
