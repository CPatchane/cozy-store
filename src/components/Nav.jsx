import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import { NavLink } from 'react-router-dom'

export const Nav = ({ t }) => (
  <nav>
    <ul className='coz-nav'>
      <li className='coz-nav-item'>
        <NavLink to='/discover' className='sto-cat-discover coz-nav-link' activeClassName='active'>
          { t('nav.discover') }
        </NavLink>
      </li>
      <li className='coz-nav-item'>
        <NavLink to='/myapps' className='sto-cat-myapps coz-nav-link' activeClassName='active'>
          { t('nav.myapps') }
        </NavLink>
      </li>
    </ul>
  </nav>
)

export default translate()(Nav)
