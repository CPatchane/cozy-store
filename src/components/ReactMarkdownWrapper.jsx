import React from 'react'
import ReactMarkdown from 'react-markdown'

export const reactMarkdownRendererOptions = {
  Link: props => <a href={props.href} target='_blank'>{props.children}</a>
}

export const ReactMarkdownWrapper = ({ source }) =>
  <ReactMarkdown
    source={source}
    renderers={reactMarkdownRendererOptions}
  />

export default ReactMarkdownWrapper
