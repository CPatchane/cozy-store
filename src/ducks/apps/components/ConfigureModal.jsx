import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import IntentModal from 'cozy-ui/react/IntentModal'
import withBreakpoints from 'cozy-ui/react/helpers/withBreakpoints'

import { getAppBySlug } from 'ducks/apps'

import { APP_TYPE } from 'ducks/apps'

export class ConfigureModal extends Component {
  constructor(props) {
    super(props)
    const { app, onNotInstalled, onWebApp } = this.props

    if (!app.installed) {
      onNotInstalled()
    }

    if (app.type === APP_TYPE.WEBAPP) {
      onWebApp()
    }
  }

  render() {
    const { app, dismissAction, onSuccess, breakpoints = {} } = this.props
    const { isMobile } = breakpoints
    return (
      <IntentModal
        action="CREATE"
        doctype="io.cozy.accounts"
        options={{ slug: app.slug }}
        dismissAction={dismissAction}
        onComplete={onSuccess}
        mobileFullscreen
        overflowHidden
        size="small"
        height={!isMobile && '35rem'}
        into="body"
      />
    )
  }
}

ConfigureModal.propTypes = {
  app: PropTypes.object.isRequired,
  dismissAction: PropTypes.func.isRequired,
  onNotInstalled: PropTypes.func.isRequired,
  onSuccess: PropTypes.func.isRequired,
  onWebApp: PropTypes.func.isRequired
}

const mapStateToProps = (state, ownProps) => ({
  app: getAppBySlug(state, ownProps.appSlug)
})

export default connect(mapStateToProps)(withBreakpoints()(ConfigureModal))
