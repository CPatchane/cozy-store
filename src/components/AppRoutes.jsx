import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

import MyApplications from '../containers/MyApplications'

const AppRoutes = () => {
  return <Switch>
    <Route path='/myapps' component={MyApplications} />
    <Redirect from='/' to='/myapps' />
  </Switch>
}

export default AppRoutes
