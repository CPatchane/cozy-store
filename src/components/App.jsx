
import React from 'react'

import Sidebar from './Sidebar'
import AppRoutes from './AppRoutes'

const App = () => (
  <div className='sto-wrapper coz-sticky'>
    <Sidebar />
    <main className='sto-content'>
      <AppRoutes />
    </main>
  </div>
)
export default App
