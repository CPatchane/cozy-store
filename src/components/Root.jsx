import React from 'react'
import { Provider } from 'react-redux'
import { I18n } from 'cozy-ui/react/I18n'
import { HashRouter } from 'react-router-dom'

import App from './App'

const Root = ({ context, lang, store }) => {
  return <I18n lang={lang} dictRequire={(lang) => require(`../locales/${lang}`)}>
    <Provider store={store}>
      <HashRouter>
        <App />
      </HashRouter>
    </Provider>
  </I18n>
}

export default Root
