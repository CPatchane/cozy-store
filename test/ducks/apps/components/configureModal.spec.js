'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { ConfigureModal } from 'ducks/apps/components/ConfigureModal'

describe('ConfigureModal component', () => {
  const konnector = {
    installed: true,
    slug: 'trinlane',
    type: 'konnector'
  }

  const props = {
    app: konnector,
    dismissAction: jest.fn(),
    onNotInstalled: jest.fn(),
    onSuccess: jest.fn(),
    onWebApp: jest.fn()
  }

  it('renders', () => {
    const component = shallow(<ConfigureModal {...props} />).getElement()
    expect(component).toMatchSnapshot()
  })

  it('calls onNotInstalled', () => {
    const uninstalledKonnector = { ...konnector, installed: false }
    const uninstalledProps = { ...props, app: uninstalledKonnector }
    shallow(<ConfigureModal {...uninstalledProps} />)
    expect(uninstalledProps.onNotInstalled.mock.calls.length).toBe(1)
  })

  it('calls onWebApp', () => {
    const webApp = { installed: true, slug: 'photos', type: 'webapp' }
    const appProps = { ...props, app: webApp }
    shallow(<ConfigureModal {...appProps} />)
    expect(appProps.onWebApp.mock.calls.length).toBe(1)
  })
})
