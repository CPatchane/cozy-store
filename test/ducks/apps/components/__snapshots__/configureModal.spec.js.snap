// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureModal component renders 1`] = `
<IntentModal
  action="CREATE"
  closable={true}
  dismissAction={[MockFunction]}
  doctype="io.cozy.accounts"
  height="35rem"
  into="body"
  mobileFullscreen={true}
  onComplete={[MockFunction]}
  options={
    Object {
      "slug": "trinlane",
    }
  }
  overflowHidden={true}
  primaryType="regular"
  secondaryType="secondary"
  size="small"
/>
`;
